# Task#1:

## Pandas

The following exercises will help you practice key functions using the pandas library.

### Question 1: import the pandas library

In [1]:
##Your code here

In [2]:
import pandas as pd

### Question 2: 

Import the `exercises.csv` file in a pandas DataFrame

In [ ]:
##Your code here

In [3]:
df = pd.read_csv('/Users/rathanrajdasari/Downloads/DS&ML/exercises.csv')
df

Gender       City  Income                Occupation Marital_Status  \
0            Male     London   40400                   Teacher         Single   
1          Female  Vancouver   62845       University Lecturer        Married   
2          Female   Montreal   66439                   Analyst         Single   
3          Female   Winnipeg  101547         Marketing Manager     Common Law   
4            Male    Calgary   46424                   Analyst         Single   
5            Male     London   70443         Real Estate Agent     Common Law   
6   Not Specified   Winnipeg   57983                Researcher     Common Law   
7            Male    Calgary   87062                 Professor        Married   
8          Female     London   54759                Researcher        Married   
9            Male    Toronto   97073                 Professor        Married   
10         Female    Calgary  106505                  Engineer        Married   
11  Not Specified    Calgary   66174                  Engineer     Common Law   
12  Not Specified   Montreal   59369                   Analyst         Single   
13         Female    Calgary   86825         Account Executive         Single   
14           Male   Winnipeg   61500           Account Manager         Single   
15           Male    Toronto   64145         Marketing Analyst     Common Law   
16           Male    Toronto  123891       Operations Director     Common Law   
17         Female   Winnipeg  107673         Investment Banker     Common Law   
18         Female    Toronto   69109                   Teacher        Married   
19           Male    Calgary   62459                Researcher        Married   
20           Male    Toronto   78712       Associate Professor     Common Law   
21           Male    Toronto   67107                Researcher         Single   
22           Male     London   83477                 Professor         Single   
23  Not Specified    Toronto  123771         Finance Executive        Married   
24  Not Specified     London   60222                Researcher     Common Law   
25         Female   Montreal   49374  Customer Service Manager         Single   
26           Male   Winnipeg   47485  Customer Service Manager         Single   

    Fixed_Monthly_Expenses  
0                     2080  
1                     2927  
2                     2811  
3                     3500  
4                     1300  
5                     2181  
6                     1423  
7                     2800  
8                     2065  
9                     3800  
10                    4000  
11                    2436  
12                    1078  
13                    2378  
14                    1200  
15                    2400  
16                    4398  
17                    3876  
18                    2987  
19                    2377  
20                    2898  
21                    1787  
22                    1199  
23                    4988  
24                    2237  
25                    1387  
26                    1203

### Question 3:

Answer the following questions using the info(), describe(), groupby(), count(), mean(), corr(), sort_values() and agg() functions:

* How many observations are there?
* What data types does the data set have?
* What is the average income in this data set?
* What is the average monthly expenses in this dataset?
* What is the average income by city in this dataset?
* Which city has the most observations?
* Which occupation has the most observations?
* What is the average income by occupation?
* What is the average monthly expenses by occupation?
* What is the average income and monthly expenses by Marital Status?
* Which people are most likely to have a high income, based on gender, occupation, city, or marital status?
* What conclusions can you make about Toronto women's income and expenses?
* Which city has the highest income?
* Calculate the correlation between income and expenses.

In [ ]:
##Your code here

In [7]:
df.shape #no of observations

(27, 6)

In [8]:
df.info() #datatypes in the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 6 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Gender                  27 non-null     object
 1   City                    27 non-null     object
 2   Income                  27 non-null     int64 
 3   Occupation              27 non-null     object
 4   Marital_Status          27 non-null     object
 5   Fixed_Monthly_Expenses  27 non-null     int64 
dtypes: int64(2), object(4)
memory usage: 1.4+ KB


In [9]:
df['Income'].mean() #average income

74176.77777777778

In [10]:
df['Fixed_Monthly_Expenses'].mean()  #avg montly expenses

2508.0

In [28]:
#average income by city in this dataset
df.groupby('City')[['Income']].mean()

Income
City                   
Calgary    75908.166667
London     61860.200000
Montreal   58394.000000
Toronto    89115.428571
Vancouver  62845.000000
Winnipeg   75237.600000

In [20]:
# city has the most observations
df['City'].mode()

0    Toronto
Name: City, dtype: object

In [22]:
#occupation has the most observations
df['Occupation'].mode()

0    Researcher
Name: Occupation, dtype: object

In [27]:
#average income by occupation
df.groupby('Occupation')[['Income']].mean()

Income
Occupation                             
Account Executive          86825.000000
Account Manager            61500.000000
Analyst                    57410.666667
Associate Professor        78712.000000
Customer Service Manager   48429.500000
Engineer                   86339.500000
Finance Executive         123771.000000
Investment Banker         107673.000000
Marketing Analyst          64145.000000
Marketing Manager         101547.000000
Operations Director       123891.000000
Professor                  89204.000000
Real Estate Agent          70443.000000
Researcher                 60506.000000
Teacher                    54754.500000
University Lecturer        62845.000000

In [26]:
#average monthly expenses by occupation
df.groupby('Occupation')[['Fixed_Monthly_Expenses']].mean()

Fixed_Monthly_Expenses
Occupation                                      
Account Executive                    2378.000000
Account Manager                      1200.000000
Analyst                              1729.666667
Associate Professor                  2898.000000
Customer Service Manager             1295.000000
Engineer                             3218.000000
Finance Executive                    4988.000000
Investment Banker                    3876.000000
Marketing Analyst                    2400.000000
Marketing Manager                    3500.000000
Operations Director                  4398.000000
Professor                            2599.666667
Real Estate Agent                    2181.000000
Researcher                           1977.800000
Teacher                              2533.500000
University Lecturer                  2927.000000

In [25]:
#average income and monthly expenses by Marital Status
df.groupby('Marital_Status')[['Income','Fixed_Monthly_Expenses']].mean()

Income  Fixed_Monthly_Expenses
Marital_Status                                      
Common Law      81198.888889             2816.555556
Married         82947.875000             3243.000000
Single          60840.000000             1642.300000

In [115]:
#people who are most likely to have a high income, based on gender, occupation, city, or marital status
df2 = df.groupby(by=['Income'])[['Gender','Occupation','City','Marital_Status']].max()
df2=df2.sort_values(by='Income',ascending= False)
df2.head(1)

Gender           Occupation     City Marital_Status
Income                                                    
123891   Male  Operations Director  Toronto     Common Law

In [51]:
#What conclusions can you make about Toronto women's income and expenses
df[(df['City'].str.startswith('T')) & (df['Gender'].str.startswith('F'))][['Income','Fixed_Monthly_Expenses']]


Income  Fixed_Monthly_Expenses
18   69109                    2987

there is only one woman working at Toronto with Income of 69109 and monthly expenses 2987

In [102]:
#Which city has the highest income?
"""nf = df.sort_values(by='Income', ascending = False)
nf.iloc[0,1]"""
df.groupby('City')['Income'].max().idxmax()

'Toronto'

In [104]:
#Calculate the correlation between income and expenses.
df[['Income','Fixed_Monthly_Expenses']].corr()

Income  Fixed_Monthly_Expenses
Income                  1.000000                0.843101
Fixed_Monthly_Expenses  0.843101                1.000000

In [105]:
df['Income'].corr(df['Fixed_Monthly_Expenses'])

0.8431007806767964